lucet-runtime/lucet-runtime-internals/src/alloc/mod.rs

```diff
@@ -79,7 +80,17 @@ pub enum AllocStrategy {
     /// supplied random number generator.
     ///
     /// This strategy is used to create reproducible behavior for testing.
-    CustomRandom(Arc<Mutex<dyn RngCore>>),
+    CustomRandom(Arc<Mutex<dyn RngCore + Send>>),
+}
+
+impl fmt::Debug for AllocStrategy {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            AllocStrategy::Linear => write!(f, "AllocStrategy::Linear"),
+            AllocStrategy::Random => write!(f, "AllocStrategy::Random"),
+            AllocStrategy::CustomRandom(_) => write!(f, "AllocStrategy::CustomRandom"),
```

Comment:
I think the convention here for the string output is `"AllocStrategy::CustomRandom(...)"` to indicate theres some elided inner part of the enum variant.

Reply:
Ooooh.  Thanks!